types/module/simulation.go

```diff
@@ -25,3 +44,35 @@ func (sm *SimulationManager) RegisterStoreDecoders() {
 		module.RegisterStoreDecoder(sm.StoreDecoders)
 	}
 }
+
+// GenerateGenesisStates generates a randomized GenesisState for each of the
+// registered modules
+func (sm *SimulationManager) GenerateGenesisStates(input *GeneratorInput) {
+	for _, module := range sm.Modules {
+		module.GenerateGenesisState(input)
+	}
+}
+
+// RandomizedSimParamChanges generates randomized contents for creating params change
+// proposal transactions
+func (sm *SimulationManager) RandomizedSimParamChanges(seed int64) {
+	r := rand.New(rand.NewSource(seed))
+
+	for _, module := range sm.Modules {
+		sm.ParamChanges = append(sm.ParamChanges, module.RandomizedParams(r)...)
+	}
+}
+
+// GeneratorInput is the input parameters used on each of the module's randomized
+// GenesisState generator function
+type GeneratorInput struct {
```

Comment:
it'd only be used on `simapp/sim_test.go`. I think we want to be explicit as @alexanderbez suggested a few times already.

Reply:
yeah I really don't see the need for a factory pattern here. We're not performing any default inits and we lose context because there's a good amount of fields.